src/js/game/hud/parts/base_toolbar.js

```diff
@@ -160,9 +160,16 @@ export class HUDBaseToolbar extends BaseHUDPart {
         }
 
         let newBuildingFound = false;
+        const direction = this.root.keyMapper.getBinding(KEYMAPPINGS.placement.rotateInverseModifier)
+            .pressed
+            ? -1
+            : 1;
         let newIndex = this.lastSelectedIndex;
-        for (let i = 0; i < this.primaryBuildings.length; ++i, ++newIndex) {
+        for (let i = 0; i < this.primaryBuildings.length; ++i, newIndex += direction) {
             newIndex %= this.primaryBuildings.length;

```

Comment:
`newIndex = safeModulo(newIndex + direction)` and remove the `++newIndex` from the for loop or ?